numpy/core/numeric.py

```diff
@@ -2089,15 +2089,12 @@ def indices(dimensions, dtype=int):
     """
     dimensions = tuple(dimensions)
     N = len(dimensions)
-    if N == 0:
-        return array([], dtype=dtype)
+    shape = (1,)*N
     res = empty((N,)+dimensions, dtype=dtype)
     for i, dim in enumerate(dimensions):
```

Comment:
Nice find. My pythonesque instincts wants me to write this like
```
for dim, r in zip(dimensions, res):
    r[:] = arange(dim, dtype=dtype).reshape(shape[:i] + (dim,) + shape[i+1:])
```
(no reason to split the line)

But really this is bikeshedding, so feel free to ignore